### PR TITLE
[Helm] Resource namespacing

### PR DESCRIPTION
**Pull request by ermik**
### Scope
Describe and implement chart resource _namespacing_ practices in appropriate for OryOS.

### Status
As described in #19 the current state of this PR shows the `noop` approach.

### CODE REVIEW

File: helm/charts/hydra/templates/configmap.yaml [hunk 1]
```diff
@@ -2,14 +2,11 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: {{ include "hydra.fullname" . }}
-  {{- if .Release.Namespace }}
```
Comment (ermik) [inline helm/charts/hydra/templates/configmap.yaml]: In this state, `helm template .`, with the `demo` flag set, yields:

```yaml
---
# Source: hydra/templates/secrets.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-name-hydra
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
type: Opaque
data:
  secretsSystem: "YS12ZXJ5LWluc2VjdXJlLXNlY3JldC1mb3ItY2hlY2tpbmctb3V0LXRoZS1kZW1v"
  secretsCookie: "YS12ZXJ5LWluc2VjdXJlLXNlY3JldC1mb3ItY2hlY2tpbmctb3V0LXRoZS1kZW1v"
  dsn: "bWVtb3J5"

---
# Source: hydra/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: release-name-hydra
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
data:
  "config.yaml": |
    secrets: {}
    serve:
      admin:
        port: 4445
      public:
        port: 4444
      tls:
        allow_termination_from:
        - 10.0.0.0/8
        - 172.16.0.0/12
        - 192.168.0.0/16
    urls:
      self: {}
    
---
# Source: hydra/templates/service-admin.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-hydra-admin
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
spec:
  type: ClusterIP
  ports:
    - port: 4445
      targetPort: http-admin
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: hydra
    app.kubernetes.io/instance: release-name

---
# Source: hydra/templates/service-public.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-hydra-public
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
spec:
  type: ClusterIP
  ports:
    - port: 4444
      targetPort: http-public
      protocol: TCP
      name: http
  selector:
    app.kubernetes.io/name: hydra
    app.kubernetes.io/instance: release-name

---
# Source: hydra/templates/tests/test-connection.yaml
apiVersion: v1
kind: Pod
metadata:
  name: "release-name-hydra-test-connection"
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
  annotations:
    "helm.sh/hook": test-success
spec:
  containers:
    - name: healthcheck-ready
      image: busybox
      command: ['wget']
      args:  ['release-name-hydra-admin:4445/health/ready']
  restartPolicy: Never

---
# Source: hydra/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-name-hydra
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: hydra
      app.kubernetes.io/instance: release-name
  template:
    metadata:
      labels:
        app.kubernetes.io/name: hydra
        app.kubernetes.io/instance: release-name
    spec:
      volumes:
        - name: hydra-config-volume
          configMap:
            name: release-name-hydra
      containers:
        - name: hydra
          image: "oryd/hydra:v1.0.0"
          imagePullPolicy: IfNotPresent
          command: ["hydra"]
          volumeMounts:
            - name: hydra-config-volume
              mountPath: /etc/config
              readOnly: true
          args: [
            "serve",
            "all",
            "--config",
            "/etc/config/config.yaml"
          ]
          ports:
            - name: http-public
              containerPort: 4444
              protocol: TCP
            - name: http-admin
              containerPort: 4445
              protocol: TCP
          livenessProbe:
            httpGet:
              path: /health/alive
              port: http-admin
          readinessProbe:
            httpGet:
              path: /health/ready
              port: http-admin
          env:
            - name: URLS_SELF_ISSUER
              value: "http://127.0.0.1:4444/"
            - name: DSN
              valueFrom:
                secretKeyRef:
                  name: release-name-hydra
                  key: dsn
            - name: SECRETS_SYSTEM
              valueFrom:
                secretKeyRef:
                  name: release-name-hydra
                  key: secretsSystem
            - name: SECRETS_COOKIE
              valueFrom:
                secretKeyRef:
                  name: release-name-hydra
                  key: secretsCookie
          resources:
            {}
            

---
# Source: hydra/templates/ingress-public.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: release-name-hydra-public
  namespace: default
  labels:
    "app.kubernetes.io/name": "hydra"
    "app.kubernetes.io/instance": "release-name"
    "app.kubernetes.io/version": "v1.0.0-rc.14_oryOS.12"
    "app.kubernetes.io/managed-by": "Tiller"
    "helm.sh/chart": "hydra-0.0.1"
spec:
  rules:
    - host: "public.hydra.localhost"
      http:
        paths:
          - path: /
            backend:
              serviceName: release-name-hydra-public
              servicePort: http

---
# Source: hydra/templates/ingress-admin.yaml


```

**Comment by ermik** [comment]
Do you want further expand this to one of the other options (2-4) from #19 or do you prefer leaving it as-is?

**Comment by aeneasr** [comment]
> Do you want further expand this to one of the other options (2-4) from #19 or do you prefer leaving it as-is?

I think it's ok for now to leave as is, but we might come back to this at a later stage!

**Comment by ermik** [comment]
This is ready.

**Comment by aeneasr** [comment]
Awesome, thank you!

**Pull request merged**